modules/sessions/components/SignUpForm.tsx

```diff
@@ -0,0 +1,232 @@
+};
+
+const validationSchema = yup
+  .object({
+    fullName: yup
+      .string()
+      .matches(
+        /^[a-zA-ZÀÁÂÃÈÉÊÌÍÒÓÔÕÙÚĂĐĨŨƠàáâãèéêìíòóôõùúăđĩũơƯĂẠẢẤẦẨẪẬẮẰẲẴẶẸẺẼỀỀỂẾưăạảấầẩẫậắằẳẵặẹẻẽềềểếỄỆỈỊỌỎỐỒỔỖỘỚỜỞỠỢỤỦỨỪễệỉịọỏốồổỗộớờởỡợụủứừỬỮỰỲỴÝỶỸửữựỳỵỷỹ\s\W|_]+$/,
+        'Tên của bạn không hợp lệ',
+      )
+      .max(40, 'Tối đa 40 ký tự')
+      .required('Bắt buộc'),
+    username: yup
+      .string()
+      .matches(
+        /^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._@]+(?<![_.])$/, // Không bắt đàu bằng . hoặc _, không chứa ._ _. __ .. ở giữa, không kêt thúc bằng _ hoặc ., chỉ chứa a-zA-Z0-9._@
+        'Tên tài khoản không hợp lệ',
+      )
+      .min(6, 'Tổi thiểu 6 ký tự')
+      .max(40, 'Tối đa 40 ký tự')
+      .required('Bắt buộc'),
+    email: yup.string().email('Email không hợp lệ').required('Bắt buộc'),
+    password: yup
+      .string()
+      .matches(
+        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/,
+        'Mật khẩu phải chứa ít nhất 8 ký tự, trong đó có phải có 1 chữ in thường, 1 chữ in hoa, 1 chữ số, và 1 ký tự đặt biệt',
+      )
+      .min(8, 'Tối thiểu 8 ký tự')
+      .max(40, 'Tối đa 40 ký tự')
+      .required('Bắt buộc'),
+    confirmPassword: yup
+      .string()
+      .oneOf([yup.ref('password'), null], 'Mật khẩu không khớp')
+      .required('Bắt buộc'),
+  })
+  .required();
+
+const SignUpForm = () => {
+  const router = useRouter();
+
+  const resolver = useYupValidateionResolver(validationSchema);
+
+  const { control, handleSubmit } = useForm<FormValues>({
+    mode: 'onChange',
+    defaultValues: {
+      fullName: '',
+      username: '',
+      email: '',
+      password: '',
+      confirmPassword: '',
+    },
+    resolver,
+  });
+
+  const onSubmit = (values: FormValues) => {
+    const payload = omit(['comfirmPassword'])(values);
+    router.push('/verify-account');
+  };
+
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        height: '100%',
+      }}
+    >
+      <Box
+        sx={{
+          margin: '2rem 10vw',
+        }}
+      >
+        <Typography mb={2} variant="h4">
+          Đăng ký
+        </Typography>
+
+        <Box mb={5} sx={{ display: 'flex' }}>
+          <Typography sx={{ mr: 1 }} variant="body2">
+            Đã có tài khoản?
+          </Typography>
+          <Link variant="subtitle2" href="/sign_in">
+            Đăng nhập tại đây
+          </Link>
+        </Box>
+
+        <Controller
+          control={control}
+          name="fullName"
+          render={({ field, fieldState: { error } }) => (
+            <TextField
+              fullWidth
+              label="Họ và tên *"
+              {...field}
+              error={Boolean(error?.message)}
+              helperText={error?.message}
+              sx={{ mb: 3 }}
+            />
+          )}
+        />
+
+        <Controller
+          control={control}
+          name="username"
```

Comment:
Did you check user input an email to this field, should we allow them to to input their email to `username` field?

Reply:
Yes, the regex is `^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._@]+(?<![_.])$` (including @). Could you access the record?

Reply:
oke cool, i could access but didn't see you typed an email to `username` field and made the screen show error